paymentsheet/src/main/java/com/stripe/android/paymentsheet/analytics/DefaultEventReporter.kt

```diff
@@ -455,6 +466,18 @@ internal class DefaultEventReporter @Inject internal constructor(
         }
     }
 
+    @Suppress("UnusedPrivateMember")
```

Comment:
Why is this unused? I don't think we should merge this until it's used.

Reply:
It's coming on this [follow-up PR](https://github.com/stripe/stripe-android/pull/10556/files) - is it fine adding it here so we can keep the related `AnalyticsV2` components used?

Reply:
We typically recommend not adding code unless it's used. Unless there's a good reason not to do it, we probably should do it.

Reply:
Main reason here is to split up changes in reviewable PRs: If we remove this, we'd need to remove all related fields that will become unused (`analyticsRequestV2Executor`, `analyticsRequestV2Factory`, `CLIENT_ID`, `ORIGIN`) to later include them in the follow up PR, making it harder to review.

Maybe adding a TODO linking the corresponding [JIRA task](https://jira.corp.stripe.com/browse/LINK_MOBILE-108) tackled in the follow-up PR works?

Reply:
TODOs in the code are even worse 🙈 Any TODOs in our codebase don't get done. At least that's the pattern I've seen.

In general, I'd rather see a larger PR that does something useful, than a smaller one that does nothing. So my 2 cents is to merge the follow up with this one.

If you disagree, I'm peaceful with disagreeing and committing here. I will approve if your choice is to merge them separately.

Reply:
went ahead and merged the other PR here!

Reply:
Can you remove the unused annotation now?